### ISSUE TITLE
Increasingly slow/unresponsive Dashboard

### ISSUE THREAD

**Issue by Clause16** [node-red/node-red-dashboard#206]
Hi guys,

I just started using node red to display the data from my sensorTag unit. Initially, the dashboard runs fine when the sensortag connected. However, as time goes by, the gauge/chart/dashboard starts to become sluggish and updates only once in a minute by the 5th minute of operation. Eventually, the dashboard gets unresponsive within just 10 minutes of operation. I would like to request for some assistance here. I've attached a picture of my flow and dashboard for your reference. 

![node red dashboard1](https://cloud.githubusercontent.com/assets/26266503/24986851/3727ebb4-202f-11e7-80aa-aa5f5d5dac12.jpg)

Additional formation:
The gauges/charts under the Air Q tab gets data and updates every 1 minute, and can run without any problem on its own if the sensortag is not connected. 

From the debug node, we can see that the sensortag node is pushing data constantly every second. This remains true for the whole duration of operation. (When the dashboard gets slow/unresponsive, the sensortag is still getting and pushing real-time data. This data can be used to turn a separate appliance on/off depending on the value, even when the dashboard isn't updating.)

Adding a delay before the gauge/chart nodes doesn't seem to help.

I'm running it on a Raspberry PI 3.

Thanks in advance for your help! Do let me know if further information is required.

**Comment by Paul-Reed** [comment]
Not sure how adding delay nodes would reduce the system demands, but have you tried slowing down the update rate by aggregating the data input.
By using an 'aggregator' node instead of a 'delay' node (especially for the chart nodes), you could average the data feed over for example 5 seconds. This would then then redraw the graph every 5 seconds instead of every second, which may help overcome this issue.

https://flows.nodered.org/node/node-red-contrib-aggregator

**Comment by dceejay** [comment]
similar / duplicate of #144

**Comment by Clause16** [comment]
Thank you for the suggestion to add in a aggregator, things seems to be working fine now!

**Comment by zioalex** [comment]
Thanks for the suggestion. Not delay any more. Thanks also to  jaggr2 for the plugin.

**Comment by QAnders** [comment]
For anyone stumbling upon this and are running on a Raspberry PI/Beaglebone, etc. with limited memory (e.g. 1 GB for the Raspberry PI), run your Node-red with flag `--node-args="--max-old-space-size=256"` to limit the memory the Node-red can munch!
(And use the Delay Node to slow down any sensors publishing more often than once a second!)
(In newer Node-RED versions there is a "Delay" node...)

It runs perfectly after adding the memory restriction!

**Comment by vortex314** [comment]
I observe the same sluggishness of the UI, on Raspberry Pi 3 the max-old-space size is already set by default in the config : /lib/systemd/system/nodered.service 
Nice=5
Environment="PI_NODE_OPTIONS=--max_old_space_size=256"
So not sure what to do else. The debug output is quick and following the data easily. The dashboard is running behind. After restarting node-red it runs fine again. Unknown root cause

**Comment by fritserasmus** [comment]
I am running my Node-RED on UBUNTU 16.04.
I tried to install 
In the instructions on:
https://www.npmjs.com/package/node-red-contrib-aggregator
I have to issue the command:

Installation
In your Node-RED directory, execute:

> npm install node-red-contrib-aggregator

Doing so I get a load of ERR messages. Probably because I am not "in the Node-RED Directory" Where is that directory to be found? Any help will be appreciated!!

**Comment by knolleary** [comment]
@fritserasmus rather than ask a new question against an unrelated issue that has been closed for over a year, please ask on the [forum](https://discourse.nodered.org) or [slack](https://nodered.org/slack).

**Comment by Snottelling** [comment]
> For anyone stumbling upon this and are running on a Raspberry PI/Beaglebone, etc. with limited memory (e.g. 1 GB for the Raspberry PI), run your Node-red with flag `--node-args="--max-old-space-size=256"` to limit the memory the Node-red can munch!
> (And use the Delay Node to slow down any sensors publishing more often than once a second!)
> (In newer Node-RED versions there is a "Delay" node...)
> 
> It runs perfectly after adding the memory restriction!

thankyou so much QAnders -- i am running mosquitto and node-red on an old android6 phone thru Termux , worked well at 1st, but as i added more sensors my dashboard was taking 5mins to load, an losing connection often. ---
with the memory restriction  "run your Node-red with flag `--node-args="--max-old-space-size=256" "
I am now powering along with a nice responsive dashboard, loading a couple of seconds... your knowledge is awesome. tyvm for sharing it
